### ISSUE TITLE
Raw response promises

### ISSUE THREAD

**Issue by luisrpfigueiredo** [microsoftgraph/msgraph-sdk-javascript#141]
When accessing the graph API response, we can pass a third option to the callback function in order to be able get a raw response, granting access to the headers.

Is there a way to do so through promises? If not, it would be nice to have, as using callbacks is... painful. A potential solution would be to have a .setResponseType('raw') or similar option available.

Edit: I found the .responseType functionality. Passing in MicrosoftGraph.ResponseType.RAW, or the hardcoded string "raw" is not working. Has this been implemented yet? ResponseType.RAW is not defined.

For reference, I'm trying to get the location header of the graph response after accessing the /items/{itemId}/copy endpoint. I know the request is successful because the items are being copied, just the response is always undefined.

**Comment by muthurathinam** [comment]
@luisrpfigueiredo 
Getting rawResponse through promises is implemented and merged to the release branch and same will be available in next release. 

Refer: [Getting Raw Response](https://github.com/microsoftgraph/msgraph-sdk-javascript/blob/dev/docs/GettingRawResponse.md)

**Comment by luis-figueiredo** [comment]
Thank you for the swift reply. Is there any estimate on when this release might be?

**Comment by muthurathinam** [comment]
I don't have the exact ETA for this, as we have more stuffs lined up before next release. I can assure you that before end of this month.

**Comment by muthurathinam** [comment]
Fixed in v1.5.2.

**Comment by dancampbell691** [comment]
@luisrpfigueiredo - did you get this working with 1.5.2 or later?  I'm on 1.6 and I still get an empty response.headers.
I've tried using both the callback method with (resp, err, rawResponse) and the promise method with .responseType(ResponseType.RAW), and both beta and v1.0, but in all cases I get nothing in the headers array.  If I look at the network tab in Chrome, I can see that the response headers did come back, including the Location value that I need.  But I can't figure out how to get it returned from the GraphClient call.

**Comment by muthurathinam** [comment]
Hi @dancampbell691 Can you share me the code that you have used to make a request ?

**Comment by dancampbell691** [comment]
hi @muthurathinam. Here's the promise-based version.  This successfully returns the raw response which includes the 202, but the headers array of the response object is empty.  If I don't include the .responseType part, then the response is just an empty string.  As I noted above, Chrome Network tab shows the response headers including location.  
One more thing - I had to punt on this and just made the call directly with axios (i.e. without the client sdk), and it worked as expected.

```javascript
res = await graphClient
    .api(`drives/${this.driveId}/items/${this.attachment.FileId}/copy`)
    .responseType(ResponseType.RAW)
    .version('beta')
    .post( 
    {
        "parentReference": { 
            'driveId': this.destinationDriveId,
            'id': this.destinationStoreGraphProperties.id
            }
        , 'name': `test ingest doc${this.$moment().format('HH-m-s')}.pdf` // temp add time so no name conflict
    });
console.log(res);
```
Any ideas would be appreciated.  Thanks!

**Comment by muthurathinam** [comment]
@dancampbell691 The returned raw response is of [this](https://developer.mozilla.org/en-US/docs/Web/API/Response) type. So when you do console.log the headers and body won't get printed along with the response object. you have to use one of the [body methods](https://developer.mozilla.org/en-US/docs/Web/API/Response#Methods) to read the body and [headers methods](https://developer.mozilla.org/en-US/docs/Web/API/Headers#Methods) to work with headers.

Basically, you have to use `res.headers.get("location")` in the above snippet, to get location header from the raw response

**Comment by dancampbell691** [comment]
@muthurathinam - Thanks!  res.headers.get('location') did the trick.

It would be helpful if there were an example in the documentation that showed the entire flow of a copy, both the client call for the copy itself, then getting the location from the response, and calling the location url.

**Comment by muthurathinam** [comment]
@dancampbell691 Sounds valid, let me update the doc showing a smiliar flow which helps in working with raw response. 😊
